### DISCUSSION TITLE
W3C Trace Context support in traced calls?

### DISCUSSION THREAD

**Discussion by fightingmonk** [traceloop/openllmetry#547] [Q&A] [answered]
Howdy @nirga . I've started using traceloop to instrument langchain workflows and have this question: 
> does OpenLLMetry inject `traceparent` headers into http calls out to OpenAI? 

I assumed it would but I don't see any evidence of W3C Trace Context headers. Am I doing it wrong, or is this a FR?

I did a couple of searches and I can't find any mention of this issue.

Here's the reason this matters to me: I use a proxy in front of OpenAI that handles data provenance, PII detection, and some proprietary business-specific data tracking. It participates in distributed traces and sends Spans to my Otel collector. Linking those spans back to the application trace of course requires it get a`traceparent` header. So if I'm using a tracing solution in my application it needs to play nice and include trace context headers in any outbound network requests.

**Comment by nirga** [comment, marked as answer]
Hey @fightingmonk! Since we're using OpenTelemetry, it should play nice with the rest of your app. Reg. injecting the headers I'm not a 100% sure, but I assume it should since we're basically creating a new span for every call out to OpenAI

**Reply by fightingmonk** [reply]
Thanks @nirga, helpful info.
